index.js

```diff
@@ -380,8 +380,11 @@ function readImportedContent(
     .then(function() {
       return processor.process(newStyles)
         .then(function(newResult) {
-          newResult.warnings().forEach(function(message) {
-            result.warn(message)
+          newResult.warnings().forEach(function(warning) {
+            result.warn(warning.text, {
+              node: warning.node,
+              plugin: warning.plugin,
+            })
```

Comment:
what about just doing `result.messages = result.messages.concat(newResult.messages)` instead of this big loop ?